jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -174,19 +177,8 @@ public static Resource newResource(String resource, boolean useCaches)
                 !resource.startsWith("file:") &&
                 !resource.startsWith("jar:"))
             {
-                try
-                {
-                    // It's a file.
-                    if (resource.startsWith("./"))
-                        resource = resource.substring(2);
-                    File file = new File(resource).getCanonicalFile();
-                    return new PathResource(file);
-                }
-                catch (IOException e2)
-                {
-                    e2.addSuppressed(e);
-                    throw e2;
-                }
+                // It's likely a file/path reference.
+                return new PathResource(Paths.get(resource));
```

Comment:
This might be a better assumption, but for 9.4 I think we really don't want to change how we work out what type a resource is as we really don't want to change behaviour... even if it is slightly wrong.  We should only fix real bugs.

Reply:
The code that was removed was a left-over from the FileResource days.

The use of getCanonicalFile() is one of the core issues from the Issue and must be addressed.
On windows, it throws for many reasons, during testing I found that it throws if the File (or any sub-path it references) doesn't exist or it contains characters in the list of unacceptable characters (such as the "*" in "path\to\dir\*").  Note: these 2 reasons are not the only reasons getCanonicalFile throws.   (On a typical Linux environment I have been unable to produce a combination of conditions that would cause it to throw)

The stripping of the "./" prefix is something that the `Path.normalize()` takes care of.
The stripped form of `resource` that the old lines 180-181 produces never exits this block of code, you'll have a PathResource or an exception.